app/src/main/java/com/example/sopt_seminar/viewmodel/SignInViewModel.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.sopt_seminar.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.sopt_seminar.data.model.User
+import com.example.sopt_seminar.data.repository.UserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.util.regex.Pattern
+import javax.inject.Inject
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class SignInViewModel @Inject constructor(
+    private val userRepository: UserRepository
+) : ViewModel() {
+    private val _user = MutableLiveData(User())
+    private val user: LiveData<User> = _user
+
+    suspend fun checkInput(idText: String, passwordText: String): Int {
+        viewModelScope.launch {
+            userRepository.getUser().collect {
+                _user.value = it
+            }
+        }
```

Comment:
예를 들어서, 
```suggestion
private val _answer = MutableLiveData<Int>()
private val answer : LiveData<Int> = _answer
// 이렇게 먼저 선언하고,
        viewModelScope.launch {
            userRepository.getUser().collect {
                _user.value = it
            }
            
            val answer = when {
            idText.isEmpty() -> 0
            passwordText.isEmpty() -> 1
            !Pattern.matches("^[a-zA-Z0-9]*\$", idText) -> 2
            !Pattern.matches(
                "^(?=.*[a-zA-Z0-9])(?=.*[a-zA-Z!@#\$%^&*])(?=.*[0-9!@#\$%^&*]).{8,15}\$",
                passwordText
            ) -> 3
            (user.value!!.userId == idText && user.value!!.userPassword == passwordText) -> 5
            else -> 4
        }
        
        _answer.value = answer
        }
```

이런식으로 한 다음에, `answer`라는 라이브데이터를 뷰에 가져가서 옵저브하는 방식으로하면 delay를 쓰지않더라도 순서가 보장될것같아요 :)

Reply:
참고해서 수정했습니다!!